### DISCUSSION TITLE
sympy.physics.mechanics / holonomic constraints/dae

### DISCUSSION THREAD

**Discussion by Peter230655** [sympy/sympy#24419] [Q&A]
I try do model a system with a configuration constraints say a four link pendulum, and I want to use scikits.odes.dae to solve it.
How do I set up Kane's equations of motion:

1.
I convert the holonomic constraint into speed constraints, and I use them in velocity constraints when setting up KanesMethod.
Then I use the speed constraints to replace the dependent velocities in the force vector by the independent velocities.

_or_

2. 
I set up Kane's equations as if there were no constraints.
Then I use the speed constraints to replace the dependent speeds by the independent speeds

Which method, if any, is correct?
My gut feeling is, that 1. is correct - but my feeling has been wrong before.

Thanks for any help!!

**Comment by moorepants** [comment]
I am not seeing a difference in 1 and 2.

**Comment by Peter230655** [comment]
Dear Jason,

If I use velocity_constraints=[…..] in setting up Kane‘s equations, the
mass matrix MM and the force seem different compared to setting up the same
system but without speed constraints. ( at least the number of operations
are different)

If I want to use dae, I must eliminate the dependent generalized speeds in
the force vector, using the equations I get from solving the speed
constraint equations I get from the configuration constraint.

Hence I wonder, which mass matrix / force vector is to be used for dae?
( My feel is, it is 1., as dae must assume the system is set up using the
speed constraints(?) )

Thanks & Merry Christmas,

Peter





On Fri 23. Dec 2022 at 17:51 Jason K. Moore ***@***.***>
wrote:

> I am not seeing a difference in 1 and 2.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/sympy/sympy/discussions/24419#discussioncomment-4487486>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQUQTNTEBWEB23M4KQUDWOXKC3ANCNFSM6AAAAAATHQXJ7Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
Best regards,

Peter Stahlecker
